mesonbuild/backend/backends.py

```diff
@@ -140,6 +140,11 @@ def get_target_filename_for_linking(self, target):
             return os.path.join(self.get_target_dir(target), link_lib)
         elif isinstance(target, build.StaticLibrary):
             return os.path.join(self.get_target_dir(target), target.get_filename())
+        elif isinstance(target, build.Executable):
+            if target.import_filename:
+                return os.path.join(self.get_target_dir(target), target.get_import_filename())
+            else:
+                return None
         raise AssertionError('BUG: Tried to link to something that\'s not a library')
```

Comment:
Could you also change this assertion to: `'BUG: Tried to link to {!r} which is not linkable'.format(target)`